SpiceSharp/Sparse/Matrix.cs

```diff
@@ -0,0 +1,426 @@
+            if (row != col || (elt = Diag[row]) == null)
+            {
+                // We have to find the element or create it!
+                elt = CreateElement(row, col);
+            }
+            return elt;
+        }
+
+        /// <summary>
+        /// Find an element in the matrix without creating it
+        /// </summary>
+        /// <param name="matrix">Matrix</param>
+        /// <param name="row">Row</param>
+        /// <param name="col">Column</param>
+        /// <returns></returns>
+        public MatrixElement FindElement(int row, int col)
+        {
+            if (row < 0 || col < 0)
+                throw new SparseException("Index out of bounds");
+
+            if (row == 0 || col == 0)
+                return TrashCan;
+
+            // Translate external indices to internal indices
+            Translation.Translate(this, ref row, ref col);
+
+            // Find the element at the right place
+            MatrixElement elt = FirstInCol[col];
+            while (elt != null)
+            {
+                if (elt.Row < row)
+                {
+                    // Next one maybe?
+                    elt = elt.NextInCol;
+                }
+                else if (elt.Row == row)
+                {
+                    // Found it!
+                    return elt;
+                }
+                else
+                {
+                    // Aww...
+                    return null;
+                }
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Create a new element in the matrix if it doesn't exist
+        /// Only used internally!
+        /// </summary>
+        /// <param name="matrix">Matrix</param>
+        /// <param name="row">Row</param>
+        /// <param name="col">Column</param>
+        /// <returns></returns>
+        private MatrixElement CreateElement(int row, int col)
+        {
+            MatrixElement elt = FirstInCol[col], last = null;
+
+            // Splice into the column vector while also searching for an existing element
+            if (elt == null || elt.Row > row)
+            {
+                // There are no elements yet in the column
+                elt = new MatrixElement(row, col);
+                elt.NextInCol = FirstInCol[col];
+                FirstInCol[col] = elt;
+                if (row == col)
+                    Diag[row] = elt;
+
+                if (RowsLinked)
+                    SpliceInRows(elt);
+            }
+            else
+            {
+                // Find the insert point
+                while (elt != null && elt.Row < row)
+                {
+                    last = elt;
+                    elt = elt.NextInCol;
+                }
+
+                // If the element does not exist yet, create it
+                if (elt == null || elt.Row != row)
+                {
+                    elt = new MatrixElement(row, col);
+                    elt.NextInCol = last.NextInCol;
```

Comment:
@svenboulanger Can last be null?

Reply:
`last` is used to find the last non-null element before the insertion point. It should not be possible for `last` to be null.
```c#
// Find the insert point
while (elt != null && elt.Row < row)
{
    last = elt;
    elt = elt.NextInCol;
}
```
Note that this code is not run if `elt == null`.